sys/cheri/cheric.h

```diff
@@ -211,23 +211,6 @@ cheri_bytes_remaining(const void * __capability cap)
 #define cheri_cap_to_typed_ptr(cap, type)				\
 	(type *)cheri_cap_to_ptr(cap, sizeof(type))
 
-#ifdef __CHERI_PURE_CAPABILITY__
-#define _CHERI_PRINTF_CAP_ARG(ptr)	(ptr)
```

Comment:
I think it may be valuable to keep a macro for a capability printf format string. Not necessarily PRIxCAPPTR, but I feel like being able to pass `void* __capability` arguments in hybrid/purecap/nocheri without ifdefs would be useful (e.g. below).

Reply:
Yeah, though there are surprisingly few cases of that.

Reply:
I'm not sure there will really be many in practice either.  Most of the time pointers are just printed via `%p` for non-cheri anyway rather than cast to an integer to be printed in a fixed format.